packages/block-library/src/breadcrumbs/edit.js

```diff
@@ -124,18 +119,20 @@ export default function BreadcrumbEdit( {
 			</div>
 		);
 	}
-	// TODO: this should be handled better when we add more types.
+
+	// Determine breadcrumb type for accurate previews (matching PHP logic).
 	let breadcrumbsType;
-	const isSpecificSupportedTypeSet = [
-		'postWithAncestors',
-		'postWithTerms',
-	].includes( type );
-	if ( isSpecificSupportedTypeSet ) {
-		breadcrumbsType = type;
+	if ( ! isPostTypeHierarchical ) {
+		breadcrumbsType = 'postWithTerms';
+	} else if ( ! postTypeHasTaxonomies ) {
+		// Hierarchical post type without taxonomies can only use ancestors.
+		breadcrumbsType = 'postWithAncestors';
 	} else {
-		breadcrumbsType = isPostTypeHierarchical
-			? 'postWithAncestors'
-			: 'postWithTerms';
+		// For hierarchical post types with taxonomies, use the attribute if valid.
+		const supportedTypes = [ 'postWithAncestors', 'postWithTerms' ];
+		breadcrumbsType = supportedTypes.includes( postBreadcrumbsType )
+			? postBreadcrumbsType
+			: 'postWithAncestors';
```

Comment:
This is a bit odd. Do we expect the value to be corrupted? Why is this better than 

```js
type = attrs.type ?? 'postWithAncestors'
```